### PR TITLE
added Mati step error structure

### PR DESCRIPTION
**Pull request by southernBuryatian**

### CODE REVIEW

File: src/models/VerificationResource.ts [hunk 1]
```diff
@@ -33,11 +33,34 @@ export enum StepStatusTypes {
   Complete = 200,
 }
 
+enum StepErrorTypes {
+  systemError = 'SystemError',
+  legacyError = 'LegacyError',
+  stepError = 'StepError'
```
Comment (Ashhm) [inline src/models/VerificationResource.ts]: Let's use PascalCase as ts recommend

Comment (southernBuryatian) [reply src/models/VerificationResource.ts]: thanks for the note! fixed

File: src/models/VerificationResource.ts [hunk 2]
```diff
@@ -33,11 +33,34 @@ export enum StepStatusTypes {
...
+
+enum ErrorCodes {
+  LegacyError = 'legacy.error',
+  SystemServiceUnavailableError = 'system.serviceUnavailable',
+  SystemInternalError = 'system.internalError',
+  StepIneNotEnoughParamsError = 'ine.notEnoughParams',
+  StepIneNotFoundError = 'ine.notFound',
+  StepCurpInvalidError = 'curp.invalid',
+  StepCurpNotFoundError = 'curp.notFound',
+  StepfullNameMismatchError = 'curp.fullNameMismatch',
```
Comment (vedi) [inline src/models/VerificationResource.ts]: `StepfullNameMismatchError` - fix case here, please

File: src/models/VerificationResource.ts [hunk 3]
```diff
@@ -33,11 +33,34 @@ export enum StepStatusTypes {
...
+  SystemError = 'SystemError',
+  LegacyError = 'LegacyError',
+  StepError = 'StepError'
+}
+
+enum ErrorCodes {
+  LegacyError = 'legacy.error',
+  SystemServiceUnavailableError = 'system.serviceUnavailable',
+  SystemInternalError = 'system.internalError',
+  StepIneNotEnoughParamsError = 'ine.notEnoughParams',
```
Comment (vedi) [inline src/models/VerificationResource.ts]: Do we need `Step` prefix here? All these error are about steps in general, and if a developer looking for an error, the step name is primary.

File: src/models/VerificationResource.ts (hunk 2 above)
Comment (Ashhm) [reply src/models/VerificationResource.ts]: Probably it should be CurpFullNameMismatch

File: src/models/VerificationResource.ts (hunk 3 above)
Comment (southernBuryatian) [reply src/models/VerificationResource.ts]: split these things to three enums

File: src/models/VerificationResource.ts (hunk 2 above)
Comment (southernBuryatian) [reply src/models/VerificationResource.ts]: fixed

**Pull request merged**